### DISCUSSION TITLE
How do I add bookmarks tab next to CPUTab

### DISCUSSION THREAD

**Discussion by wulfcare** [x64dbg/x64dbg#3375] [Q&A]
I have tried editing the .ini file and adding BookmarksTab to [TabOrder] but it doesn't actually seem to add to the panel.

[TabOrder]
BreakpointsTab=4
CPUTab=0
CallStackTab=5
HandlesTab=C
LogTab=1
MemoryMapTab=3
NotesTab=2
ReferencesTab=A
SEHTab=6
ScriptTab=7
SourceTab=9
SymbolsTab=8
ThreadsTab=B
TraceTab=D
BookmarksTab=E

Is it actually possible to add it, so its at the top panel each time I load x64dbg.

**Comment by mrfearless** [comment]
The order placed in the TabOrder section doesnt matter. What does effect the ordering is the number each entry has. So starting with 0 to E (hexadecimal for 14)
So `BookmarksTab=0`,  `CPUTab=1`, and so on till you reach the end of your ordering and each entry has a unique number representing the order index. 

Edit: the Bookmarks tab doesn't actually have an entry itself. Likely you would need to add the `ReferencesTab` as the first entry. But the references tab will only populate once the bookmark or label or functions button is clicked or the appropriate menu item is selected. References is the catchall for content that is generated. It is not updated dynamically, so any changes require re-generating the content: bookmarks, labels, etc etc.

**Reply by mrexodia** [reply]
Unlike IDA the bookmarks/labels functions are not dynamically updated as @mrfearless mentioned. Best you can do right now is set up a system breakpoint script and run the `bookmarklist` command there.
